src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+                           Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+
+    if (this->unk_1F0 && limbIndex == 0xB) {
+        Matrix_InsertXRotation_s(this->unk_1D8.y, 1);
+    }
+    if ((this->unk_1EC == 9 || this->unk_1EC == 8) && limbIndex == 0xA) {
+        *dList = 0;
+    }
+    return 0;
+}
+
+void EnOsn_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+    Vec3f sp30 = D_80AD2574;
+    Vec3s sp28 = D_80AD2580;
+
+    if (limbIndex == 0xB) {
+        Matrix_MultiplyVector3fByState(&sp30, &actor->focus.pos);
+    }
+    if (((this->unk_1EC == 0x11) || (this->unk_1EC == 0x15) || (this->unk_1EC == 0x16)) && (limbIndex == 6)) {
+        Matrix_StatePush();
+        Matrix_InsertTranslation(-400.0f, 1100.0f, -200.0f, 1);
+        Matrix_InsertXRotation_s(sp28.x, 1);
+        Matrix_RotateY(sp28.y, 1);
+        Matrix_InsertZRotation_s(sp28.z, 1);
+
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList((*gfx)++, &object_osn_DL_0192A0);
+        Matrix_StatePop();
+    }
+}
+
+void EnOsn_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    GraphicsContext* temp_s0;
+    EnOsn* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (this->unk_1FC == 0xFF) {
+        func_8012C28C(globalCtx->state.gfxCtx);
+        if ((this->unk_1EC == 0xB) || (this->unk_1EC == 0xC) || (this->unk_1EC == 0x17) ||
+            (globalCtx->msgCtx.unk11F04 == 0x1FCA)) {
+            gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80AD258C));
+            gSPSegment(POLY_OPA_DISP++, 0x09, Lib_SegmentedToVirtual(D_80AD2594));
+        } else if ((this->unk_1EC == 7) || (this->unk_1EC == 3) || (this->unk_1EC == 0xD)) {
+            gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80AD2590));
+            gSPSegment(POLY_OPA_DISP++, 0x09, Lib_SegmentedToVirtual(D_80AD2598));
+        } else {
+            gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80AD2588));
+            gSPSegment(POLY_OPA_DISP++, 0x09, Lib_SegmentedToVirtual(D_80AD2594));
+        }
+        gDPSetEnvColor(POLY_OPA_DISP++, 255, 255, 255, 255);
+        Scene_SetRenderModeXlu(globalCtx, 0, 1);
+        POLY_OPA_DISP = SkelAnime_DrawFlex(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable,
+                                           this->skelAnime.dListCount, EnOsn_OverrideLimbDraw, EnOsn_PostLimbDraw,
+                                           &this->actor, POLY_OPA_DISP);
+    } else {
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        gSPSegment(POLY_XLU_DISP++, 0x08, Lib_SegmentedToVirtual(D_80AD2588));
+        gSPSegment(POLY_XLU_DISP++, 0x09, Lib_SegmentedToVirtual(D_80AD2594));
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 255, this->unk_1FC);
+        Scene_SetRenderModeXlu(globalCtx, 1, 2U);
```

Comment:
```suggestion
        Scene_SetRenderModeXlu(globalCtx, 1, 2);
```
Generally we request values to leave out the explicit unsigned for literals if they aren't required to match.